cli/src/semgrep/rule_match.py

```diff
@@ -365,6 +366,19 @@ def get_match_based_key(self) -> Tuple:
                 path,
                 self.annotated_rule_name,
             )
+
+        # Check if the match is non-code-match and reachable is False
+        if "sca_info" in self.extra and not self.extra["sca_info"].reachable:
+            # Include dependency package and version in the key
+            dependency = self.extra["sca_info"].dependency_match.found_dependency
+            return (
+                match_formula_str,
+                path,
+                self.rule_id,
+                dependency.package,
+                dependency.version,
+            )
```

Comment:
Just one thing to think about: This will change `match_based_id` for existing findings, causing new ones to be created in the DB instead of updating existing rows. Will that affect any of our workflows negatively?

Reply:
That seems potentially very disruptive. We should talk to the findings team about this I think.

Reply:
Yeah, adding anything to the match_based_key is a bad idea because changing match_based_id will in fact mark the existing findings as fixed and then re-open them as new issues, losing all triage information along the way.

Instead, if the thing you want to add here is not a "secret" (i.e. doesn't need to be hidden behind a hashing function) I would recommend sending it in plaintext to the app, and handling it there instead. That also avoids the problem of this change getting rolled out slowly over time, since not everyone runs the latest version of the CLI on all scans.

Two options in the app: 
1. changing the match_based_id in a deterministic and reversible way such that you can convert back and forth between the old and new match_based_id formats
2. changing the logic to use more than just match_based_id when determining unique identifiers for supply_chain

I'm not sure which of these to recommend without a little more context on why the CLI is generating findings with the same match_based_id in the first place. Usually the index handles this problem (if all else is the same about two findings in a given scan, they get an index like `_0` and `_1` appended to them to differentiate. 

Let's find some time to chat?